packages/ee/lib/api/apiKeys.ts

```diff
@@ -0,0 +1,10 @@
+import { randomBytes, createHash } from "crypto";
+
+// Hash the API key to check against when veriying it. so we don't have to store the key in plain text.
+export const hashAPIKey = (apiKey: string): string => createHash("sha256").update(apiKey).digest("hex");
```

Comment:
The difficulty comes in at the dehashing part - createHash is one way.

Reply:
I think for what I read that salting is only needed for passwords because they are not unique, (two users can have same password), and so to avoid rainbow table attacks.
In this case, we're making it unique constraint at the db level and so thus should be not applicable.

We can add salting if we think it's needed though

Reply:
Resolving for now as it seems safe to not salt unique api keys!